### DISCUSSION TITLE
Edge optimization

### DISCUSSION THREAD

**Discussion by shvchk** [farag2/Sophia-Script-for-Windows#513] [Ideas]
There are some default Edge settings, which some might find suboptimal for privacy, performance or other reasons. Like Startup boost, running in background, new tab page with news and whatnot, telemetry, crypto wallet functions, etc.

Would be nice to include changing some of these to Sophia script and SophiApp.

One good annotated list is https://github.com/TairikuOokami/Windows/blob/main/Windows%20Tweaks.bat#L1181 and of course there is official https://learn.microsoft.com/en-us/deployedge/microsoft-edge-policies

**Comment by farag2** [comment]
Hm... I will check later. 6.15.3 is ready to be released.

**Reply by shvchk** [reply]
Ok, no hurry :wink: 
There are also [Douloureux/MS-Edge-Hardening](https://github.com/Douloureux/MS-Edge-Hardening), [Dreista/FuckEdge](https://github.com/Dreista/FuckEdge) and [melody tweaks](https://github.com/jbara2002/melody_windows/blob/master/MelodyScript.Settings/Browser%20Configuration/Edge/Microsoft%20Edge%20Tweaks.reg) — not annotated, but gives an idea what settings people are interested in changing.

**Comment by shvchk** [comment]
Gonna just leave some notes here, which might be useful for implementation of this feature. Ignore for now : )


## Settings types: mandatory and recommended

[MS docs: Configure Edge](https://learn.microsoft.com/en-us/deployedge/configure-microsoft-edge)
> Microsoft Edge supports _mandatory_ and _recommended_ policies. Mandatory policies override user preferences and prevent the user from the policy. Recommended policies provide a default setting that the user can override. Most policies are only mandatory but there's a subset that is mandatory and recommended. If both versions of a policy are set, the mandatory setting takes precedence. A recommended policy only takes effect when the user hasn't modified the setting.

Path for both mandatory and recommended is documented in [MS Edge policy docs](https://learn.microsoft.com/en-us/deployedge/microsoft-edge-policies).

E.g., for [Startup boost setting](https://learn.microsoft.com/en-us/deployedge/microsoft-edge-policies#startupboostenabled):

> Windows Registry Settings:
> - Path (Mandatory): SOFTWARE\Policies\Microsoft\Edge
> - Path (Recommended): SOFTWARE\Policies\Microsoft\Edge\Recommended
> - Value Name: StartupBoostEnabled
> - Value Type: REG_DWORD


## Determine most wanted settings

List of settings which are changed by at least 2 of 5 repos: all mentioned previously + [LeDragoX/Win-Debloat-Tools](https://github.com/LeDragoX/Win-Debloat-Tools):

<details><summary>Show</summary>

| # of 5 | Setting                                        | Description                                                                                                                                                                                |
|--------|------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| 4      | AddressBarMicrosoftSearchInBingProviderEnabled | Enable Microsoft Search in Bing suggestions in the address bar                                                                                                                             |
| 4      | BackgroundModeEnabled                          | Continue running background apps after Microsoft Edge closes                                                                                                                               |
| 4      | DiagnosticData                                 | Send required and optional diagnostic data about browser usage                                                                                                                             |
| 4      | EdgeCollectionsEnabled                         | Enable the Collections feature                                                                                                                                                             |
| 4      | EdgeShoppingAssistantEnabled                   | Shopping in Microsoft Edge Enabled                                                                                                                                                         |
| 4      | NewTabPageContentEnabled                       | Allow Microsoft News content on the new tab page                                                                                                                                           |
| 4      | NewTabPageHideDefaultTopSites                  | Hide the default top sites from the new tab page                                                                                                                                           |
| 4      | PersonalizationReportingEnabled                | Allow personalization of ads, Microsoft Edge, search, news and other Microsoft services by sending browsing history, favorites and collections, usage and other browsing data to Microsoft |
| 4      | PromotionalTabsEnabled                         | Enable full-tab promotional content                                                                                                                                                        |
| 4      | ResolveNavigationErrorsUseWebService           | Enable resolution of navigation errors using a web service                                                                                                                                 |
| 4      | ShowMicrosoftRewards                           | Show Microsoft Rewards experiences                                                                                                                                                         |
| 4      | StartupBoostEnabled                            | Enable startup boost                                                                                                                                                                       |
| 3      | AdsSettingForIntrusiveAdsSites                 | Ads setting for sites with intrusive ads                                                                                                                                                   |
| 3      | AllowGamesMenu                                 | Allow users to access the games menu (deprecated)                                                                                                                                          |
| 3      | DefaultGeolocationSetting                      | Default geolocation setting                                                                                                                                                                |
| 3      | DefaultJavaScriptJitSetting                    | Control use of JavaScript JIT                                                                                                                                                              |
| 3      | EnableMediaRouter                              | Enable Google Cast                                                                                                                                                                         |
| 3      | FamilySafetySettingsEnabled                    | Allow users to configure Family safety and Kids Mode                                                                                                                                       |
| 3      | HideFirstRunExperience                         | Hide the First-run experience and splash screen                                                                                                                                            |
| 3      | NewTabPagePrerenderEnabled                     | Enable preload of the new tab page for faster rendering                                                                                                                                    |
| 3      | NewTabPageQuickLinksEnabled                    | Allow quick links on the new tab page                                                                                                                                                      |
| 3      | NewTabPageSearchBox                            | Configure the new tab page search box experience                                                                                                                                           |
| 3      | PasswordMonitorAllowed                         | Allow users to be alerted if their passwords are found to be unsafe                                                                                                                        |
| 3      | SearchSuggestEnabled                           | Enable search suggestions                                                                                                                                                                  |
| 3      | ShowRecommendationsEnabled                     | Allow feature recommendations and browser assistance notifications from Microsoft Edge                                                                                                     |
| 3      | SitePerProcess                                 | Enable site isolation for every site                                                                                                                                                       |
| 3      | UserFeedbackAllowed                            | Allow user feedback                                                                                                                                                                        |
| 3      | DNSInterceptionChecksEnabled                   | DNS interception checks enabled                                                                                                                                                            |
| 3      | EdgeFollowEnabled                              | Enable Follow service in Microsoft Edge                                                                                                                                                    |
| 3      | HubsSidebarEnabled                             | Show Hubs Sidebar                                                                                                                                                                          |
| 3      | SpotlightExperiencesAndRecommendationsEnabled  | Choose whether users can receive customized background images and text, suggestions, notifications, and tips for Microsoft services                                                        |
| 2      | ApplicationGuardFavoritesSyncEnabled           | Application Guard Favorites Sync Enabled                                                                                                                                                   |
| 2      | ApplicationGuardTrafficIdentificationEnabled   | Application Guard Traffic Identification                                                                                                                                                   |
| 2      | AudioSandboxEnabled                            | Allow the audio sandbox to run                                                                                                                                                             |
| 2      | AutofillAddressEnabled                         | Enable AutoFill for addresses                                                                                                                                                              |
| 2      | AutofillCreditCardEnabled                      | Enable AutoFill for payment instruments                                                                                                                                                    |
| 2      | AutomaticHttpsDefault                          | Configure Automatic HTTPS                                                                                                                                                                  |
| 2      | BingAdsSuppression                             | Block all ads on Bing search results                                                                                                                                                       |
| 2      | BrowserSignin                                  | Browser sign-in settings                                                                                                                                                                   |
| 2      | ClearCachedImagesAndFilesOnExit                | Clear cached images and files when Microsoft Edge closes                                                                                                                                   |
| 2      | ConfigureShare                                 | Configure the Share experience                                                                                                                                                             |
| 2      | DefaultSensorsSetting                          | Default sensors setting                                                                                                                                                                    |
| 2      | DownloadRestrictions                           | Allow download restrictions                                                                                                                                                                |
| 2      | FavoritesBarEnabled                            | Enable favorites bar                                                                                                                                                                       |
| 2      | ForceSync                                      | Force synchronization of browser data and do not show the sync consent prompt                                                                                                              |
| 2      | InPrivateModeAvailability                      | Configure InPrivate mode availability                                                                                                                                                      |
| 2      | InternetExplorerIntegrationLevel               | Configure Internet Explorer integration                                                                                                                                                    |
| 2      | LocalBrowserDataShareEnabled                   | Enable Windows to search local Microsoft Edge browsing data                                                                                                                                |
| 2      | LocalProvidersEnabled                          | Allow suggestions from local providers                                                                                                                                                     |
| 2      | PasswordGeneratorEnabled                       | Allow users to get a strong password suggestion whenever they are creating an account online                                                                                               |
| 2      | PasswordManagerEnabled                         | Enable saving passwords to the password manager                                                                                                                                            |
| 2      | PasswordRevealEnabled                          | Enable Password reveal button                                                                                                                                                              |
| 2      | PromptForDownloadLocation                      | Ask where to save downloaded files                                                                                                                                                         |
| 2      | QuickViewOfficeFilesEnabled                    | Manage QuickView Office files capability in Microsoft Edge                                                                                                                                 |
| 2      | ShowCastIconInToolbar                          | Show the cast icon in the toolbar                                                                                                                                                          |
| 2      | TranslateEnabled                               | Enable Translate                                                                                                                                                                           |
| 2      | AlternateErrorPagesEnabled                     | Suggest similar pages when a webpage can't be found                                                                                                                                        |
| 2      | CryptoWalletEnabled                            | Enable CryptoWallet feature                                                                                                                                                                |
| 2      | EdgeEnhanceImagesEnabled                       | Enhance images enabled                                                                                                                                                                     |
| 2      | MicrosoftEdgeInsiderPromotionEnabled           | Microsoft Edge Insider Promotion Enabled                                                                                                                                                   |
| 2      | QuickSearchShowMiniMenu                        | Enables Microsoft Edge mini menu                                                                                                                                                           |
| 2      | ShowPDFDefaultRecommendationsEnabled           | Allow notifications to set Microsoft Edge as default PDF reader                                                                                                                            |
| 2      | SmartScreenEnabled                             | Configure Microsoft Defender SmartScreen                                                                                                                                                   |
| 2      | StandaloneHubsSidebarEnabled                   | Standalone Sidebar Enabled                                                                                                                                                                 |
| 2      | TabServicesEnabled                             | Tab Services enabled                                                                                                                                                                       |
| 2      | VisualSearchEnabled                            | Visual search enabled                                                                                                                                                                      |
| 2      | SleepingTabsEnabled                            | Configure sleeping tabs                                                                                                                                                                    |
| 2      | WebWidgetIsEnabledOnStartup                    | Allow the Search bar at Windows startup                                                                                                                                                    |
| 2      | DirectInvokeEnabled                            | Allow users to open files using the DirectInvoke protocol                                                                                                                                  |
| 2      | TrackingPrevention                             | Block tracking of users' web-browsing activity                                                                                                                                             |
| 2      | VerticalTabsAllowed                            | Configures availability of a vertical layout for tabs on the side of the browser                                                                                                           |
| 2      | ExperimentationAndConfigurationServiceControl  | Control communication with the Experimentation and Configuration Service                                                                                                                   |
| 2      | NetworkPredictionOptions                       | Enable network prediction                                                                                                                                                                  |
| 2      | BackgroundTemplateListUpdatesEnabled           | Enables background updates to the list of available templates for Collections and other features that use templates (deprecated)                                                           |
| 2      | SpellcheckEnabled                              | Enable spellcheck                                                                                                                                                                          |
| 2      | WebWidgetAllowed                               | Enable the Search bar                                                                                                                                                                      |
| 2      | WPADQuickCheckEnabled                          | Set WPAD optimization                                                                                                                                                                      |
| 2      | DefaultFileSystemReadGuardSetting              | Control use of the File System API for reading                                                                                                                                             |
| 2      | DefaultFileSystemWriteGuardSetting             | Control use of the File System API for writing                                                                                                                                             |
| 2      | DefaultInsecureContentSetting                  | Control use of insecure content exceptions                                                                                                                                                 |
| 2      | DefaultNotificationsSetting                    | Default notification setting                                                                                                                                                               |
| 2      | DefaultWebBluetoothGuardSetting                | Control use of the Web Bluetooth API                                                                                                                                                       |
| 2      | DefaultWebHidGuardSetting                      | Control use of the WebHID API                                                                                                                                                              |
| 2      | DefaultWebUsbGuardSetting                      | Control use of the WebUSB API                                                                                                                                                              |
| 2      | SmartScreenDnsRequestsEnabled                  | Enable Microsoft Defender SmartScreen DNS requests                                                                                                                                         |
| 2      | NewTabPageAllowedBackgroundTypes               | Configure the background types allowed for the new tab page layout                                                                                                                         |
| 2      | TyposquattingCheckerEnabled                    | Configure Edge TyposquattingChecker                                                                                                                                                        |
| 2      | SiteSafetyServicesEnabled                      | Allow users to configure Site safety services                                                                                                                                              |

</details>

Those changed by 3+ of 5 are probably the most wanted settings : ) Might be useful to cherry-pick some from 2 of 5 part of the list, too.

UPD: added settings description, extracted from [MS Edge policy docs](https://learn.microsoft.com/en-us/deployedge/microsoft-edge-policies). See [code in Jupyter Notebook](https://gist.github.com/shvchk/9c55be4f584183ed25fcec3d82fd43fb) if interested to reuse for other settings.

## Misc

Extracted all lines mentioning Edge from [TairikuOokami/Windows](https://github.com/TairikuOokami/Windows/blob/main/Windows%20Tweaks.bat) (might be good for reference): https://gist.github.com/shvchk/f05c0bdcdde2071b7cad223bb5785bed

---

There is also an option to install extensions by default, using [Edge Addon ID](https://microsoftedge.microsoft.com), e.g. uBlock Origin:
```
[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Edge\ExtensionInstallForcelist]
"1"="odfafepnkmbhccpbejgmiehpchacaeak"
```

**Comment by farag2** [comment]
I thought out and do not consider adding any Edge related settings. Sophia Script is for Windows )).

**Reply by shvchk** [reply]
Well, Edge is part of Windows now : ) Its settings, like auto-start in the background, will impact every Windows user by default.

But that's your call. Hope you reconsider at some point. Keep up the good work :muscle:

**Reply by farag2** [reply]
I've no desire to configure it. Probably, because I'll never use it. Also, adding that tons of features will make the script is even heavier.

**Reply by shvchk** [reply]
Me too. But that's the thing: even if you don't use it, it will still start automatically and run in the background by default, if you started it even once (which I think most people do, at least to download another browser :smile: ). That "feature" is controlled by `HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Edge\StartupBoostEnabled` (0/1).

**Comment by shvchk** [comment]
Since this probably won't be in Sophia script any time soon, I've created [edge-please.reg](https://github.com/shvchk/edge-please) for those who still want this. It changes 15 most irritating Edge settings by default, has 50+ more optional (not applied by default, uncomment to your liking). Every setting has a description and a link to official docs.
